pkg/controller/namespace/deletion/namespaced_resources_deleter.go

```diff
@@ -135,10 +136,13 @@ func (d *namespacedResourcesDeleter) Delete(nsName string) error {
 	// there may still be content for us to remove
 	estimate, err := d.deleteAllContent(namespace.Name, *namespace.DeletionTimestamp)
 	if err != nil {
+		d.updateStatusCondition(namespace.Name, err)
 		return err
 	}
 	if estimate > 0 {
-		return &ResourcesRemainingError{estimate}
+		est := &ResourcesRemainingError{estimate}
```

Comment:
Instead of making two calls, maybe consider making a list of conditions that you want to set, and then when you get here, make a single call that sets them all at once?